From dbfa93dcb0ec42b2847031ecffe59dcb8fa3c430 Mon Sep 17 00:00:00 2001
From: Varun <vyaparla@habana.ai>
Date: Wed, 16 Feb 2022 08:17:07 +0200
Subject: [PATCH] [SW-69281] common: elbi: rename elbi related functions

This change renames the elbi backend related functions
to generic name instead of the type of memory used for
elbi backend.
In gaudi, spsram is used as a backend storage for fit
via elbi and in gaudi2, arc farm dccm is used.

Change-Id: I7c8b50ce660e999afb8ce8ad8aeecb45a9b3a41e
---

diff --git a/app/preboot/preboot.c b/app/preboot/preboot.c
index 3cc17a2..4e83daf 100644
--- a/app/preboot/preboot.c
+++ b/app/preboot/preboot.c
@@ -275,8 +275,8 @@
 		enum preboot_mem_access_mode mode)
 {}
 __weak void preboot_security_dynamic_img_sram_close(void) {}
-__weak void preboot_security_dynamic_img_spsram_open(void) {}
-__weak void preboot_security_dynamic_img_spsram_close(void) {}
+__weak void preboot_security_dynamic_img_elbi_reg_open(void) {}
+__weak void preboot_security_dynamic_img_elbi_reg_close(void) {}
 __weak void preboot_security_cleanup_before_run(void) {}
 __weak void preboot_led_on(void) {}
 __weak void preboot_sp_sram_init(void) {}
diff --git a/arch/gaudi/arm/security.c b/arch/gaudi/arm/security.c
index 48e06d8..0411f9a 100644
--- a/arch/gaudi/arm/security.c
+++ b/arch/gaudi/arm/security.c
@@ -2078,7 +2078,7 @@
 	}
 }
 
-void preboot_security_dynamic_img_spsram_close(void)
+void preboot_security_dynamic_img_elbi_reg_close(void)
 {
 	struct preboot_sec_dyn_pcie_rr_cfg *cfg =
 					&preboot_sec_dyn_cfg[SPSRAM_IMAGE];
@@ -2089,7 +2089,7 @@
 	preboot_secure_pcie_rr(cfg->rr, cfg->min, cfg->max);
 }
 
-void preboot_security_dynamic_img_spsram_open(void)
+void preboot_security_dynamic_img_elbi_reg_open(void)
 {
 	preboot_secure_pcie_rr(preboot_sec_dyn_cfg[SPSRAM_IMAGE].rr, 0, 0);
 }
diff --git a/include/security.h b/include/security.h
index 571ba13..ea1f6f4 100644
--- a/include/security.h
+++ b/include/security.h
@@ -24,8 +24,8 @@
 void preboot_security_dynamic_img_sram_close(void);
 void preboot_security_dynamic_img_sram_open(enum mem_region img_region,
 					    enum preboot_mem_access_mode mode);
-void preboot_security_dynamic_img_spsram_close(void);
-void preboot_security_dynamic_img_spsram_open(void);
+void preboot_security_dynamic_img_elbi_reg_close(void);
+void preboot_security_dynamic_img_elbi_reg_open(void);
 void preboot_security_cleanup_before_run(void);
 void preboot_security_sp_sram_write_key_hash(u8 *key, size_t len);
 void preboot_security_hbm_pci_open_ro(void);
diff --git a/lib/preboot/module_fwupd.c b/lib/preboot/module_fwupd.c
index 4bfddf3..f85e554 100644
--- a/lib/preboot/module_fwupd.c
+++ b/lib/preboot/module_fwupd.c
@@ -429,7 +429,7 @@
 	module_fwupd_reset_state(state);
 	preboot_set_sram_sf_access(false);
 	preboot_security_dynamic_img_sram_close();
-	preboot_security_dynamic_img_spsram_close();
+	preboot_security_dynamic_img_elbi_reg_close();
 }
 
 void module_fwupd_pcie_disable(bool is_pcie_disabled)
@@ -472,7 +472,7 @@
 			preboot_security_dynamic_img_sram_open(state->mregion,
 							       ACCESS_RW);
 		else
-			preboot_security_dynamic_img_spsram_open();
+			preboot_security_dynamic_img_elbi_reg_open();
 	}
 
 	return sts;
@@ -677,7 +677,7 @@
 	 */
 	preboot_set_sram_sf_access(false);
 	preboot_security_dynamic_img_sram_close();
-	preboot_security_dynamic_img_spsram_close();
+	preboot_security_dynamic_img_elbi_reg_close();
 
 	err = module_fwupd_load_type(state);
 
diff --git a/lib/preboot/module_lkd.c b/lib/preboot/module_lkd.c
index ee3ebf9..5f612e2 100644
--- a/lib/preboot/module_lkd.c
+++ b/lib/preboot/module_lkd.c
@@ -456,7 +456,7 @@
 
 	preboot_set_sram_sf_access(false);
 	preboot_security_dynamic_img_sram_close();
-	preboot_security_dynamic_img_spsram_close();
+	preboot_security_dynamic_img_elbi_reg_close();
 
 	err = lkd_get_image_addr(&desc);
 	if (err) {
@@ -559,7 +559,7 @@
 
 	preboot_security_dynamic_img_sram_open(MEM_LKD_PCI, ACCESS_RW);
 	if (sts.is_elbi_desc)
-		preboot_security_dynamic_img_spsram_open();
+		preboot_security_dynamic_img_elbi_reg_open();
 
 	return err;
 }
